ss/pebbledb/db.go

```diff
@@ -556,27 +594,60 @@ func (db *Database) RawImport(ch <-chan types.RawSnapshotNode) error {
 		}
 
 		var counter int
+		var latestKey []byte // store the latest key from the batch
+		var latestModule string
 		for entry := range ch {
 			err := batch.Set(entry.StoreKey, entry.Key, entry.Value, entry.Version)
 			if err != nil {
 				panic(err)
 			}
 
+			latestKey = entry.Key // track the latest key
+			latestModule = entry.StoreKey
 			counter++
+
 			if counter%ImportCommitBatchSize == 0 {
+				startTime := time.Now()
+
+				// Commit the batch and record the latest key as metadata
 				if err := batch.Write(); err != nil {
 					panic(err)
 				}
 
+				// Persist the latest key in the metadata
+				if err := db.SetLatestMigratedKey(latestKey); err != nil {
+					panic(err)
+				}
+
+				if err := db.SetLatestMigratedModule(latestModule); err != nil {
+					panic(err)
+				}
+
+				fmt.Printf("Time taken to write batch counter %d: %v\n", counter, time.Since(startTime))
```

Comment:
Will this log be too annoying?

Reply:
Also I wonder if there's a way to track the current progress of how many modules has been migrated till now?